src/main/java/ch/njol/skript/conditions/CondHasPassengers.java

```diff
@@ -0,0 +1,71 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import org.bukkit.entity.Entity;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+@Name("Has Passengers")
+@Description("Checks whether the given entities have passengers.")
+@Examples("if the vehicle has passengers:")
+@Since("INSERT VERSION")
+public class CondHasPassengers extends PropertyCondition<Entity> {
+
+	static {
+		Skript.registerCondition(CondHasPassengers.class,
+				"%entities% (has|have) [a[ny]|:no] passenger[s]",
+				"%entities% (doesn't|does not|do not|don't) have [a[ny]] passenger[s]"
```

Comment:
I don't think `zombie do not have any passenger` should be accepted

Reply:
This is prominent on all already existing property conditions. You cannot account for every proper plurality. It's for `all of the zombies do not have any passengers`